src/ChainRec.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var assert = require('./internal/assert');
+var chain = require('./internal/chain');
+var chainRec = require('./internal/chainRec');
+var map = require('./internal/map');
+
+
+module.exports = function(equals, CR) {
+  return {
+
+    equivalence: assert.forall1(function(i) {
+      var p = '???';
+      var n = '???';
+      var d = '???';
+      var lhs = chainRec(CR)(function(next, done, v) { return p(v) ? map(done)(d(v)) : map(next)(n(v)); })(i);
+      var rhs = (function step(v) { return p(v) ? d(v) : chain(step)(n(v)); }(i));
+      return equals(lhs, rhs);
+    })
```

Comment:
This variables stand for:
- **p**-redicate (to detect if we are done)
- calculate-**n**-extValue (if we are not done, this calculates next value for the iteration)
- calculate-**d**-oneValue (if we are done calculate done value and this will be result of whole iteration)
- **i**-nitialValue (initial value of the iteration)

It's used like this [in FL/test.js](https://github.com/fantasyland/fantasy-land/blob/master/test.js#L58-L64). 

Different monads might use different variation of arguments so I think yes, they should be provided by caller.